ui/app/AppLayouts/Profile/views/ChangePasswordView.qml

```diff
@@ -109,19 +120,34 @@ SettingsContentBase {
                     StatusButton {
                         text: biometricsSwitch.checked ? qsTr("Yes, enable biometrics") : qsTr("Yes, disable biometrics")
                         onClicked: {
-                            if (biometricsSwitch.checked && !biometricsSwitch.biometricsEnabled) {
-                                biometricsSwitch.requestForEnabling = true;
-                                root.privacyStore.tryStoreToKeyChain();
-                            } else if (!biometricsSwitch.checked) {
-                                biometricsSwitch.requestForEnabling = false;
-                                root.privacyStore.tryRemoveFromKeyChain();
+                            if (biometricsSwitch.checked) {
+                                root.privacyStore.tryStoreToKeyChain()
```

Comment:
Is it calling the `privacyStore` in order to keep compatibility between old/new onboarding? Or am I missing sth? If it's a case, I think some comment would be useful there.

Reply:
To save password to Keychain, we actually need to get the password from the user first (we don't store in memory it ourselves).
`tryStoreToKeyChain` is the method that will invoke the Authentication popup and eventually emit the `saveBiometricsRequested` signal.

The whole thing about authentication popup is very confusing, we need to refactor that for sure.
But for now it does the job in exactly the same manner as it works everywhere else in the app, where authentication is needed.